soroban-env-host/src/test/map.rs

```diff
@@ -372,3 +369,182 @@ fn large_map_exceeds_budget() {
         (Instant::now() - start).as_secs_f64()
     );
 }
+
+#[test]
+fn initialization_invalid() -> Result<(), HostError> {
+    use crate::EnvBase;
+    let host = observe_host!(Host::default());
+
+    // Out of order keys
+    assert!(host
+        .map_new_from_slices(&["b", "a"], &[1u32.into(), 2u32.into()])
+        .is_err());
+
+    // Duplicate
+    assert!(host
+        .map_new_from_slices(&["a", "a"], &[1u32.into(), 2u32.into()])
+        .is_err());
+
+    // large map from slices
+    let buf = vec![0; 7_000_000];
+    let scv_bytes = ScVal::Bytes(buf.try_into().unwrap());
+    let bytes_val = host.to_host_val(&scv_bytes)?;
+
+    let vals = vec![bytes_val; 5];
+    let keys = ["a", "b", "c", "d", "e"];
+
+    let map = host.map_new_from_slices(&keys, &vals.as_slice())?;
```

Comment:
Should we also have large values as keys for completeness?

Reply:
The keys must be symbols. 
But for completeness I've added a case where the key slice is too long.